src/main/java/ch/njol/skript/events/EvtEntityPotion.java

```diff
@@ -63,22 +63,13 @@ public boolean check(Event event) {
 		}
 
 		EntityPotionEffectEvent potionEvent = (EntityPotionEffectEvent) event;
-		if (potionEffects == null || potionEvent.getNewEffect() == null) {
-			return false;
-		}
+		boolean effectMatches = potionEffects == null || (potionEvent.getNewEffect() != null && potionEffects.check(event, effectType -> effectType.equals(potionEvent.getNewEffect().getType())));
```

Comment:
This should probably check for old effect too, since I would think removing speed would count for `potion effect modification of speed`